src/compiler/sys.ts

```diff
@@ -747,6 +747,10 @@ namespace ts {
 
                 function fileChanged(curr: any, prev: any) {
                     if (+curr.mtime === 0) {
+                        if (eventKind === FileWatcherEventKind.Deleted) {

```

Comment:
I have some concerns with this approach.  First, I think it might be cleaner (since it would be stateless) to pull up the `mtime` check from below.  Both `curr` and `prev` have timestamp 0, so it would succeed and suppress the callback.  

Second, I'm concerned that this only addresses the symptoms of the problem.  Why are we getting a fileChanged notification that has never existed and continues not to exist?  My limited debugging suggests that this used to be suppressed when `oldStatus === -1` and `newStatus === -1` (somewhere in sys or fs) but that we are now seeing `newStatus === -4058`.  If this is a Node limitation, then I think we need a big comment here explaining that we're working around a library limitation.

Reply:
My breakpoint was at line 1448 of `fs.js`, but I can't get it to load the text without debugging the issue again, which I'm not currently set up to do.

Reply:
https://github.com/nodejs/node/blob/52e869bee8f639fb2b256efa814446d0c325a4ac/lib/fs.js#L1435